09_json/runnerdave/cmd/puppy_server/main.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	puppy "github.com/anz-bank/go-course/09_json/runnerdave/pkg/puppy"
+	store "github.com/anz-bank/go-course/09_json/runnerdave/pkg/puppy/store"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	args           = os.Args[1:]
+	data           = kingpin.Flag("data", "data file").Short('d').Default("data.json").ExistingFile()
+	out  io.Writer = os.Stdout
+)
+
+func main() {
+	_, err := kingpin.CommandLine.Parse(args)
+	if err != nil {
+		fmt.Fprintf(out, "Command line could not be parsed, error: %v", err)
+		return
+	}
+
+	d, _ := ioutil.ReadFile(*data)
+	puppies, uerr := unmarshalPuppies(d)
+	if uerr != nil {
+		return
+	}
+
+	mapStore := storeInMap(puppies)
```

Comment:
This is fine for the purpose of the lab (showing you can initialize either underlying store type).

Keep in mind the for normal systems (ie: next lab), you would select the implementation via command line flag or config. Would be useful to think about how this might work.

Reply:
will do